iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -540,111 +635,187 @@ TEST_F(ServiceDiscovery_test, NonExistingServicesAreNotFound)
+                                                        SERVICE_DESCRIPTION.getEventIDString(),
+                                                        MessagingPattern::REQ_RES);
+
+    ASSERT_THAT(serviceContainerPubSub.size(), Eq(1U));
+    EXPECT_THAT(*serviceContainerPubSub.begin(), Eq(SERVICE_DESCRIPTION));
+
+    ASSERT_THAT(serviceContainerReqRes.size(), Eq(1U));
+    EXPECT_THAT(*serviceContainerReqRes.begin(), Eq(SERVICE_DESCRIPTION));
 }
 
-TEST_F(ServiceDiscovery_test, FindServiceWithEmptyCallableDoesNotDie)
+///
+/// Publisher-Subscriber
+///
+
+TEST_F(ServiceDiscoveryPubSub_test, FindServiceWithWildcardsReturnsOnlyIntrospectionServicesAndServiceRegistry)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "d944f32c-edef-44f5-a6eb-c19ee73c98eb");
+    auto searchResult = this->sut.findService(
+        iox::capro::Wildcard, iox::capro::Wildcard, iox::capro::Wildcard, MessagingPattern::PUB_SUB);
+
+
+    for (auto& service : searchResult)
+    {
+        EXPECT_THAT(service.getInstanceIDString().c_str(), StrEq("RouDi_ID"));
+    }
+
+    this->sut.findService(
+        iox::capro::Wildcard, iox::capro::Wildcard, iox::capro::Wildcard, findHandler, MessagingPattern::PUB_SUB);
+    compareAndClearServiceContainers(searchResult, searchResultOfFindServiceWithFindHandler);
+}
+
+TEST_F(ServiceDiscoveryPubSub_test, FindServiceWithEmptyCallableDoesNotDie)
 {
     ::testing::Test::RecordProperty("TEST_ID", "7e1bf253-ce81-47cc-9b4a-605de7e49b64");
     const iox::capro::ServiceDescription SERVICE_DESCRIPTION("ninjababy", "pow", "pow");
     iox::popo::UntypedPublisher publisher(SERVICE_DESCRIPTION);
     iox::cxx::function_ref<void(const ServiceDescription&)> searchFunction;
-    sut.findService(iox::capro::Wildcard, iox::capro::Wildcard, iox::capro::Wildcard, searchFunction);
+    this->sut.findService(
+        iox::capro::Wildcard, iox::capro::Wildcard, iox::capro::Wildcard, searchFunction, MessagingPattern::PUB_SUB);
 }
 
-TEST_F(ServiceDiscovery_test, ServiceDiscoveryIsAttachableToWaitSet)
+TEST_F(ServiceDiscoveryPubSub_test, ServiceDiscoveryIsAttachableToWaitSet)
 {
     ::testing::Test::RecordProperty("TEST_ID", "fc0eeb7a-6f2a-481f-ae8a-1e17460e261f");
     iox::popo::WaitSet<10U> waitSet;
 
     waitSet
-        .attachEvent(sut,
+        .attachEvent(this->sut,
                      ServiceDiscoveryEvent::SERVICE_REGISTRY_CHANGED,
                      0U,
                      iox::popo::createNotificationCallback(testCallback))
         .and_then([]() { GTEST_SUCCEED(); })
         .or_else([](auto) { GTEST_FAIL() << "Could not attach to wait set"; });
 }
 
-TEST_F(ServiceDiscovery_test, ServiceDiscoveryIsNotifiedbyWaitSetAboutSingleService)
+TEST_F(ServiceDiscoveryPubSub_test, ServiceDiscoveryIsNotifiedbyWaitSetAboutSingleService)
 {
     ::testing::Test::RecordProperty("TEST_ID", "f1cf36b5-3db2-4e6f-8e05-e7e449530ec0");
     iox::popo::WaitSet<1U> waitSet;
 
     waitSet
-        .attachEvent(sut,
+        .attachEvent(this->sut,
```

Comment:
Technically `this` is not needed as it is not a typed test but ok ...

Reply:
Please fix this on #1151